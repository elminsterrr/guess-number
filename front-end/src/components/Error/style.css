.Error-message {
  color: #dc143c;
}
