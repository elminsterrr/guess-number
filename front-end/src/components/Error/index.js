import React from 'react';
import './style.css';

const Error = props => (
  <div>
    <p className="Error-message">We are sorry, but something went wrong.</p>
    <p className="Error-message">{props.error && props.error.toString()}!</p>
  </div>
);

export default Error;
