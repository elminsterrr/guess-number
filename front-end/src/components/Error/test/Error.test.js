import React from 'react';
import { shallow } from 'enzyme';
import Error from '../';

describe('Error', () => {
  const error = shallow(<Error />);

  it('renders correctly', () => {
    expect(error).toMatchSnapshot();
  });
});
