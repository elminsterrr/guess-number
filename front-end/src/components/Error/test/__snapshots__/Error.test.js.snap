// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="Error-message"
>
          We are sorry, but something went wrong.
</p>,
        <p
          className="Error-message"
>
          !
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "We are sorry, but something went wrong.",
          "className": "Error-message",
        },
        "ref": null,
        "rendered": "We are sorry, but something went wrong.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            "!",
          ],
          "className": "Error-message",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          "!",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="Error-message"
>
            We are sorry, but something went wrong.
</p>,
          <p
            className="Error-message"
>
            !
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "We are sorry, but something went wrong.",
            "className": "Error-message",
          },
          "ref": null,
          "rendered": "We are sorry, but something went wrong.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              "!",
            ],
            "className": "Error-message",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            "!",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
